octoprint_mrbeam/analytics/uploader.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python
+
+import requests
+import time
+import os
+import sys
+import threading
+try:
+	from octoprint_mrbeam.mrb_logger import mrb_logger
+except:
+	import logging
+from octoprint_mrbeam.util.cmd_exec import exec_cmd
+
+
+TOKEN_URL           = "https://europe-west1-mrb-analytics.cloudfunctions.net/get_upload_tocken"
+UPLOAD_URL_TEMPLATE = "https://storage-upload.googleapis.com/{bucket}"
+
+
+
+
+
+class FileUploader(object):
+
+	STATUS_QUEUED =     'queued'
+	STATUS_INIT =       'init'
+	STATUS_VERIFY =     'verify'
+	STATUS_GET_TOKEN =  'get_token'
+	STATUS_UPLOAD =     'upload_file'
+	STATUS_REMOVE =     'remove_file'
+	STATUS_DONE =       'done'
+
+	FILE_UPLOAD_DELAY_AFTER_INIT_DEFAULT = 15.0
+	MIN_LIN_COUNT_TO_UPLOAD = 10
+
+
+	def __init__(self,
+	             analytics_dir,
+	             files=[],
+	             delete_on_success=False,
+	             delay=None,
+	             analytics_files_prefix=None,
+	             analytics_files_suffix=None,
+	             logger=None):
+		self._logger = logger or mrb_logger("octoprint.plugins.mrbeam.analytics.uploader")
+		self.delay = delay or self.FILE_UPLOAD_DELAY_AFTER_INIT_DEFAULT
+		self.analytics_dir = analytics_dir
+		self.analytics_files_prefix = analytics_files_prefix
+		self.analytics_files_suffix = analytics_files_suffix
+		self.files = []
```

Comment:
Should this take 'files' instead of directly its default value '[ ]'?
```suggestion
		self.files = files
```

Reply:
thx. good catch!